src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs

```diff
@@ -7569,7 +8049,7 @@ public static void Main()
     {
         if (true)
         {
-            object d1 = Dummy(new C(""a""), new C(""b"") is var x1, x1);
+            (object d1, object d2) = (Dummy(new C(""a""), (new C(""b"") is var x1), x1), x1);
```

Comment:
>object [](start = 24, length = 6)

It might be nice to have one test or two for nested deconstruction declarations, in each of the two forms `var (x, (y, z))` and `(var x, (var y, var z))`